config/main.py

```diff
@@ -542,32 +586,132 @@ def config():
 config.add_command(nat.nat)
 
 @config.command()
+@click.option('-n', '--namespace', help='Namespace name')
 @click.option('-y', '--yes', is_flag=True, callback=_abort_if_false,
-                expose_value=False, prompt='Existing file will be overwritten, continue?')
-@click.argument('filename', default='/etc/sonic/config_db.json', type=click.Path())
-def save(filename):
+                expose_value=False, prompt='Existing files will be overwritten, continue?')
+@click.argument('filename',nargs=-1)
+def save(namespace, filename):
     """Export current config DB to a file on disk."""
-    command = "{} -d --print-data > {}".format(SONIC_CFGGEN_PATH, filename)
-    run_command(command, display_cmd=True)
+    if namespace:
+        if not is_multi_asic():
+            click.echo("Namespace is not significant in a Single ASIC platform")
+            return None
+        if not validate_namespace(namespace):
+            click.echo("Invalid Namespace entered {}".format(namespace))
+            return None
+        if not len(filename):
+            inst = namespace[len(NAMESPACE_PREFIX)]
+            cfg_file = "/etc/sonic/config_db{}.json".format(inst)
+        else:
+            cfg_file = filename[0]
+
+        command = "{} -n {} -d --print-data > {}".format(SONIC_CFGGEN_PATH, namespace, cfg_file)
+        run_command(command, display_cmd=True)
+    else:
+        if not len(filename):
+            host_cfg_file = DEFAULT_CONFIG_DB_FILE
+        else:
+            host_cfg_file = filename[0]
+
+        command = "{} -d --print-data > {}".format(SONIC_CFGGEN_PATH, host_cfg_file)
+        run_command(command, display_cmd=True)
+
+        """In case of multi-asic mode we have additional config_db{NS}.json files for
+           various namespaces created per ASIC. {NS} is the namespace index.
+        """
+        if is_multi_asic():
+            num_asic = _get_num_asic()
+            for inst in range(num_asic):
+                namespace = "{}{}".format(NAMESPACE_PREFIX, inst)
+                # Get the file from user input, else take the default file /etc/sonic/config_db{NS_id}.json
+                if len(filename) > inst+1:
+                    cfg_file = filename[inst+1]
+                else:
+                    cfg_file = "/etc/sonic/config_db{}.json".format(inst)
+
+                command = "{} -n {} -d --print-data > {}".format(SONIC_CFGGEN_PATH, namespace, cfg_file)
+                run_command(command, display_cmd=True)
 
 @config.command()
+@click.option('-n', '--namespace', help='Namespace name')
 @click.option('-y', '--yes', is_flag=True)
-@click.argument('filename', default='/etc/sonic/config_db.json', type=click.Path(exists=True))
-def load(filename, yes):
+@click.argument('filename',nargs=-1)
+def load(filename, yes, namespace):
```

Comment:
check my comments on the save comments. i think they apply to this command as well.

Reply:
Take care of.